client/src/components/ClickOutComponent/ClickOutComponent.js

```diff
@@ -0,0 +1,79 @@
+/* global document */
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+/**
+ * Wrapper component that fires a handler (`onClickOut`) when a user clicks
+ * outside of the component.
+ *
+ * Example:
+ *
+ * ```jsx
+ *    class TestComponent extends React.Component {
+ *      ...
+ *
+ *      handleClickOut() {
+ *        alert('User clicked outside of TestComponent.');
+ *      }
+ *
+ *      render() {
+ *        return (
+ *          <ClickOutComponent onClickOut={this.handleClickOut} callerComponent={this}>
+ *            <div>... </div>
+ *          </ClickOutComponent>
+ *        );
+ *       }
+ *    < /ClickOutComponent>
+ *  ```
+ */
+
+class ClickOutComponent extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.wasClicked = false;
+    this.handleElementClick = this.handleElementClick.bind(this);
+    this.handleDocumentClick = this.handleDocumentClick.bind(this);
+  }
+
+  componentDidMount() {
+    const el = ReactDOM.findDOMNode(this);
+    el.addEventListener('click', this.handleElementClick);
+    document.addEventListener('click', this.handleDocumentClick);
+
+    el.addEventListener('touchstart', this.handleElementClick);
+    document.addEventListener('touchstart', this.handleDocumentClick);
+  }
+
+  componentWillUnmount() {
+    const el = ReactDOM.findDOMNode(this);
+    el.removeEventHandler('click', this.handleElementClick);
+    el.removeEventHandler('click', this.handleDocumentClick);
+  }
+
+  handleElementClick() {
+    this.wasClicked = true;
+  }
+
+  handleDocumentClick() {
+    if (!this.wasClicked) {
+      this.props.onClickOut.call(this.props.callerComponent);
+    } else {
```

Comment:
maybe remove the `else` so that `wasClicked` can be repeatedly used?
At the moment, it looks like once `wasClicked` goes true, it stays true.